### DISCUSSION TITLE
Recursive upload from web interface

### DISCUSSION THREAD

**Discussion by johannSc** [immich-app/immich#3552] [Q&A] [answered]
Hello,

This project is handsome, congrats!
I have try upoad photos from my iphone, with succes.

But my principal pictures directory is from my Synology: (photo/year/month)

I haven't found how upload a directory and his subdirectory, from my mac I must move on a directory and select all pictures for upload them for each month so. 

If it's possible, could you explain to me how upload a entire directory year?

An other question, there are somes files (MOC and HEIC) in the upload directory, but they didn't be delete, is it normal?

Thanks for your help

**Comment by alextran1502** [comment, marked as answer]
Hello, you should use the CLI tool for bulk import, the web doesn't work very well if you try to import thousand of photos at a time 

https://immich.app/docs/features/bulk-upload

**Reply by johannSc** [reply]
Ok I will try with the CLI tool. And for my question about the files into the upload directory, any idea?

**Reply by alextran1502** [reply]
There is a flag of `--recursive` to upload the nested folders

**Reply by alextran1502** [reply]
<img width="717" alt="image" src="https://github.com/immich-app/immich/assets/27055614/4d7af840-9c67-4e62-bd86-a6bba12cc9b3">

**Comment by johannSc** [comment]
thanks for your help, well done.
